addon/serializers/web-api.js

```diff
@@ -21,11 +22,13 @@ export default DS.RESTSerializer.extend({
 
   serializeHasMany: function(snapshot, json, relationship) {
     let key = this.payloadKeyFromModelName(relationship.key);
-    json[key] = [];
+    if (this._shouldSerializeHasMany(snapshot, key, relationship)) {
```

Comment:
Good catch. Any idea whether this is a new method or something that can be used with legacy code? It might make sense to include this with the other pre-1.13.8 change I mentioned (if it gets broken apart).